packaging/build_rpm.py

```diff
@@ -336,22 +338,46 @@ def install_mock():
 
 
 def check_mock_config():
-    # allow network access during the build
-    # (we have to download packages from pypi)
-    mock_config = check_output(
-        MOCK + ['--debug-config',
-        ]).splitlines()
-
-    for line in mock_config:
-        if all(s in line for s in ('rpmbuild_networking', 'True')):
-            break
-    else:
-        logger.info('configuring rpmbuild_networking')
-        check_call(
-                '''echo "config_opts['rpmbuild_networking'] = True" '''
-                '| sudo -n tee -a /etc/mock/site-defaults.cfg',
-                shell=True,
-                )
+    """
+    allow network access during the build and set the %dist macro
+    (we have to download packages from pypi)
+    """
+    mock_config = check_output(MOCK + ['--debug-config', '-qq'])
+
+    config_opts = {}
+    config = {'config_opts': config_opts}
+    exec mock_config in config, config
```

Comment:
Can you explain again the need for the `exec` statement? I’m not sure I’m following exactly what/why you’re doing.

Reply:
this is because mock config files are python code, so the way to parse mock config files is to use `exec`. This is unfortunate but that's a design choice by mock, and they are doing the same as well: https://github.com/rpm-software-management/mock/blob/7f363b1c15d5cc8e893fdf495713515df94876c7/mock/py/mockbuild/util.py#L1182